### PR TITLE
[AWS] [CloudWatch] Add dimensions metadata

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

- Add dimension fields to the cloudwatch data_stream
- Add ingest pipeline to generate `aws.dimensions.fingerprint` field
- remove `aws.s3.bucket.name` field

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## How to test this PR locally

- Related https://github.com/elastic/integrations/issues/6293

## Screenshots

Mapping:
Screenshot 2023-07-05 at 18 07 01

Testing with [TSDB-migration-test-kit](https://github.com/elastic/TSDB-migration-test-kit)

<details>

```
python3.10 main.py
You're testing with version 8.8.0.

Testing data stream metrics-aws.cloudwatch_metrics-default.
Index being used for the documents is .ds-metrics-aws.cloudwatch_metrics-default-2023.07.05-000001.
Index being used for the settings and mappings is .ds-metrics-aws.cloudwatch_metrics-default-2023.07.05-000001.

The time series fields for the TSDB index are: 
        - dimension (5 fields):
                - agent.id
                - aws.dimensions.fingerprint
                - aws.s3.bucket.name
                - cloud.account.id
                - cloud.region
        - routing_path (5 fields):
                - agent.id
                - aws.dimensions.fingerprint
                - aws.s3.bucket.name
                - cloud.account.id
                - cloud.region

Index tsdb-index-enabled successfully created.

Copying documents from .ds-metrics-aws.cloudwatch_metrics-default-2023.07.05-000001 to tsdb-index-enabled...
All 204 documents taken from index .ds-metrics-aws.cloudwatch_metrics-default-2023.07.05-000001 were successfully placed to index tsdb-index-enabled.
```

</details>

### CODE REVIEW

**Comment by constanca-m** [comment]
Quick question: why is aws.s3.bucketname a dimension? Isn't a dimension fingerprint enough?

**Comment by tetianakravchenko** [comment]
> Quick question: why is aws.s3.bucketname a dimension? Isn't a dimension fingerprint enough?


@constanca-m with the default [configuration for `ec2:instance`](https://github.com/elastic/integrations/blob/main/packages/aws/data_stream/cloudwatch_metrics/manifest.yml#L47-L61) I couldn't actually get this field in document. It is as well not available in the [sample_event](https://github.com/elastic/integrations/blob/main/packages/aws/data_stream/cloudwatch_metrics/sample_event.json)

I assumed it is only used for the S3 metrics. As for the S3 bucket name is not always part of `aws.dimentions.*`:
```
    "aws": {
      "s3": {
        "bucket": {
          "name": "tkravchenko-test"
        }
      },
      "cloudwatch": {
        "namespace": "AWS/S3"
      },
      "s3_request": {
        "latency": {
          "total_request": {
            "ms": 2.5
          }
        },
        "requests": {
          "head": 2,
          "total": 2
        },
        "downloaded": {
          "bytes_per_period": 730,
          "bytes": 365
        },
        "errors": {
          "4xx": 1,
          "5xx": 0
        }
      },
      "dimensions": {
        "FilterId": "Screenshot"
      }
```
I've added this field.
@elastic/obs-cloud-monitoring could you please clarify it? why this field is declared in cloudwatch data_stream explicitly? Is there any specific reason? Or it should be removed from the fields definition?

**Review by constanca-m — APPROVED** [review]
> I assumed it was added explicitly for s3 use case, https://github.com/orgs/elastic/teams/obs-cloud-monitoring could you please clarify it? why this field is declared in cloudwatch data_stream explicitly? Is there any specific reason? Or it should be removed from the fields definition?

Doesn't sound like a blocker to TSDB then. Even if it is an extra field, that is something that can be changed along the way.

File: packages/aws/data_stream/cloudwatch_metrics/fields/package-fields.yml [hunk 1]
```diff
@@ -7,12 +7,18 @@
         Tag key value pairs from aws resources.
     - name: s3.bucket.name
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/cloudwatch_metrics/fields/package-fields.yml:8]: I think `aws.s3.bucket.name` is added here by mistake. We can remove it. I don't see `cloudwatch_metrics` data stream collects metrics with this dimension.

Comment (tetianakravchenko) [reply packages/aws/data_stream/cloudwatch_metrics/fields/package-fields.yml]: @kaiyan-sheng I've removed this field - https://github.com/elastic/integrations/pull/6827/commits/493b9a2f5897618b17560da60894d2b3f877a0c0, please have another look

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**